MobileCenterDistribute/MobileCenterDistribute/MSDistribute.m

```diff
@@ -214,23 +214,32 @@ - (void)requestInstallInformationWith:(NSString *)releaseHash {
     if (url) {
 
 /*
- * Only iOS 9.x and 10.x will download the update after users click the "Install" button. 
+ * Only iOS 9.x and 10.x will download the update after users click the "Install" button.
  * We need to force-exit the application for other versions or for any versions when the update is mandatory.
  */
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wpartial-availability"
       Class clazz = [SFSafariViewController class];
-#pragma clang diagnostic pop
-      /*
-       * TODO Checking operating system version is a workaround for iOS 11 where SFSafariViewController can't read Safari's cookies.
-       * Revert this change when SFAuthenticationSession will be ready.
-       */
       if (clazz && ![NSProcessInfo.processInfo isOperatingSystemAtLeastVersion:(NSOperatingSystemVersion){11, 0, 0}]) {
 
-        // Manipulate App UI on the main queue.
-        dispatch_async(dispatch_get_main_queue(), ^{
-          [self openURLInEmbeddedSafari:url fromClass:clazz];
-        });
+        Class authClazz = [SFAuthenticationSession class];
+
+        // iOS 11
+        if (authClazz) {
+
+          // Manipulate App UI on the main queue.
+          dispatch_async(dispatch_get_main_queue(), ^{
+            [self openURLInAuthenticationSessionWith:url fromClass:authClazz];
+          });
+        } else {
+          // iOS 9 and 10
+
+          // Manipulate App UI on the main queue.
```

Comment:
Can you just combine the two comments like `Manipulate App UI on the main queue on iOS 11`, `Manipulate App UI on the main queue on iOS 9 and 10`?
Or, you can just remove `// Manipulate App UI on the main queue.` since it can be easily realized that it is on main queue.
`// iOS 11` should be in if statement, `// iOS9 and 10` should have a line break above.